File: Interview Preparation Kit - CPP/02. Arrays/004. Minimum Swaps 2.cpp
// Problem: https://www.hackerrank.com/challenges/minimum-swaps-2/problem
// Score: 40


#include <iostream>
#include <vector>
using namespace std;


int main(){
    int n;
    cin >> n;
    vector<int> arr;

    for (int i = 0; i < n; i++){
        int tmp;
        cin >> tmp;
        arr.push_back(tmp - 1);
    }

    int i = 0;
    int ans = 0;
    while (i < n){
        if (arr[i] != i){
            swap(arr[i], arr[arr[i]]);
            ans++;
        }
        else{
            i++;
        }
    }

    cout << ans;

    return 0;
}

File: Interview Preparation Kit - Python/02. Arrays/004. Minimum Swaps 2.py
# Problem: https://www.hackerrank.com/challenges/minimum-swaps-2/problem
# Score: 40


n = int(input())
arr = list(map(int, input().split()))
count = 0

i = 0
while i < len(arr):
    if arr[i] != i + 1:
        arr[arr[i] - 1], arr[i] = arr[i], arr[arr[i] - 1]
        count += 1
    else:
        i += 1

print(count)
